src/main/java/com/edmunds/rest/databricks/DTO/scim/user/UserDTO.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2018 Edmunds.com, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.edmunds.rest.databricks.DTO.scim.user;
+
+import com.edmunds.rest.databricks.DTO.scim.group.GroupDTO;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Databricks SCIM user.
+ * @see <a href="https://tools.ietf.org/html/rfc7643#section-4.1">https://tools.ietf.org/html/rfc7643#section-4.1</a>
+ */
+public class UserDTO {
+
+  @JsonProperty("schemas")
+  private final String[] schemas = new String[]{"urn:ietf:params:scim:schemas:core:2.0:User"};
+  @JsonProperty("id")
+  private long id;
+  @JsonProperty("userName")
+  private String userName;
+  @JsonProperty("name")
+  private NameDTO name;
+  @JsonProperty("emails")
+  private EmailDTO[] emails;//can't be actually customized. setting username will create an entry (work,username,primary)
+  @JsonProperty("active")
+  private boolean active = true;
+  @JsonSerialize(using = UserGroupSerializer.class)
+  @JsonDeserialize(using = UserGroupDeSerializer.class)
+  private GroupDTO[] groups = new GroupDTO[0];
+  @JsonProperty("entitlements")
+  private EntitlementsDTO[] entitlements = new EntitlementsDTO[0];
+  @JsonProperty("displayName")
+  private String displayName;
+
+  public UserDTO() {
+  }
+
+  public UserDTO(UserDTO from) {
+    this.id = from.id;
+    this.userName = from.userName;
+    if (from.name != null) {
+      this.name = new NameDTO(from.name);
+    }
+    if (from.emails != null) {
```

Comment:
Please try some of this array copying with Arrays.copy methods.

Reply:
+1